src/notification.c

```diff
@@ -140,12 +154,15 @@ activate_action (GDBusConnection *connection,
                               object_path,
                               "org.freedesktop.Application",
                               "Activate",
-                              g_variant_new ("(@a{sv})",
-                                             g_variant_builder_end (&pdata)),
+                              g_variant_new ("(@a{sv})", pdata),
                               NULL,
                               G_DBUS_CALL_FLAGS_NONE,
                               -1, NULL, NULL, NULL);
 
+      /* The application may not implement the `org.freedesktop.Application`, so
+       * add the platform data also to the `ActionInvoked` signal */
+      g_variant_builder_add (&parms, "v", pdata);
```

Comment:
i think we should add only add the token once instead of adding `desktop-startup-id` and `activation-token`. Since this isn't part of the xdg-desktop-portal specs (for now) nothing would expect `desktop-startup-id`